components/suggestion/suggestion_box/use_container_ref.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {QuickInputState} from '../../quick_input';
+
+import * as UserAgent from 'utils/user_agent';
+
+import {ISuggestionBoxProps, ISuggestionBoxState} from './types';
+import {SuggestionHandler} from './use_suggestion_handler';
+
+export const useContainerRef = (
```

Comment:
Ideally we would move away from using refs as SuggestionBox heavily relies on them at the moment. Handling focus with the SuggestionBox is weird, and we have to use focusin/focusout to get access to e.relatedTarget. React 17 is going to be switching the event handlers it uses internally, so we won't need the refs once that comes out. But if we can remove the need for them all together it will be great :)

Reply:
Lemme check about this.